manager/src/main/java/com/redhat/service/bridge/manager/api/user/ProcessorsAPI.java

```diff
@@ -37,6 +38,7 @@
 
 @Path(APIConstants.USER_API_BASE_PATH)
 @Produces(MediaType.APPLICATION_JSON)
+@Transactional
 public class ProcessorsAPI {
```

Comment:
`@Transactional` moved to the entry point of the application as Quarkus guide recommends. In this way, when the ProcessorResponse is built by Jackson it can `LAZY` fetch the filters

Reply:
Problem will be that with `list` methods, this leads to a n+1 select on the DB e.g:

- select the list of Processors
- foreach Processor, select the list of filters

Instead I'd suggest we do a `left join fetch`  in the queries that service these methods when loading our data from the DB. So the query would be something like:

`from Processor p join fetch p.bridge left join fetch p.filters where p.bridge.id=:bridgeId and p.bridge.customerId=:customerId`

This will mean that the Filters collection is eagerly initialised within the single DB query. By using the `left join fetch` in the queries where we need it, we can:

- Avoid the N+1 selects
- Keep the `@Transaction` boundary on the `ProcessorService`
- Keep the Filter collection marked as `fetch = FetchType.LAZY` on the Entity, so that we don't load it by default, unless you use a NamedQuery or call `getFilters()` within a Transaction.

Reply:
@robpblake done, now the queries look like
```java
@NamedQueries({
        @NamedQuery(name = "PROCESSOR.findByBridgeIdAndName",
                query = "from Processor p left join fetch p.filters where p.name=:name and p.bridge.id=:bridgeId"),
        @NamedQuery(name = "PROCESSOR.findByStatus",
                query = "from Processor p join fetch p.bridge left join fetch p.filters where p.status in (:statuses) and p.bridge.status='AVAILABLE'"),
        @NamedQuery(name = "PROCESSOR.findByIdBridgeIdAndCustomerId",
                query = "from Processor p join fetch p.bridge left join fetch p.filters where p.id=:id and (p.bridge.id=:bridgeId and p.bridge.customerId=:customerId)"),
        @NamedQuery(name = "PROCESSOR.findByBridgeIdAndCustomerId",
                query = "from Processor p join fetch p.bridge left join fetch p.filters where p.bridge.id=:bridgeId and p.bridge.customerId=:customerId"),
        @NamedQuery(name = "PROCESSOR.countByBridgeIdAndCustomerId",
                query = "select count(p.id) from Processor p where p.bridge.id=:bridgeId and p.bridge.customerId=:customerId"),
        @NamedQuery(name = "PROCESSOR.idsByBridgeIdAndCustomerId",
                query = "select p.id from Processor p where p.bridge.id=:bridgeId and p.bridge.customerId=:customerId order by p.submittedAt asc"),
        @NamedQuery(name = "PROCESSOR.findByIds",
                query = "select p from Processor p join fetch p.bridge left join fetch p.filters where p.id in (:ids)")
})
```